catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -244,6 +300,68 @@ private CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
         actionRegistry);
   }
 
+  private void attachFileWithTransformer(
+      ServiceReference<QueryResponseTransformer> queryResponseTransformer,
+      CqlQueryResponse cqlQueryResponse,
+      Request request,
+      Response response) {
+    response.status(500);
+    BinaryContent content;
+    try {
+      content =
+          bundleContext
+              .getService(queryResponseTransformer)
+              .transform(cqlQueryResponse.getQueryResponse(), new HashMap<>());
+    } catch (CatalogTransformerException e) {
+      LOGGER.debug(
+          "Error fetching binary content for transformer with id: {}",
+          queryResponseTransformer.getProperty("id"),
+          e);
+      return;
+    }
+
+    String mimeType = (String) queryResponseTransformer.getProperty("mime-type");
+    MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+    String fileExt;
+    try {
+      fileExt = allTypes.forName(mimeType).getExtension();
+    } catch (MimeTypeException e) {
+      LOGGER.debug("Error fetching file extension for mime-type: {}", mimeType, e);
+      return;
+    }
+
+    String acceptEncoding = request.headers(HttpHeaders.ACCEPT_ENCODING);
+
+    boolean shouldGzip =
+        StringUtils.isNotBlank(acceptEncoding) && acceptEncoding.toLowerCase().contains("gzip");
+
+    response.type(mimeType);
+    String attachment =
+        String.format("attachment;filename=export-%s%s", Instant.now().toString(), fileExt);
+    response.header("Content-Disposition", attachment);
+
+    try {
+      try (OutputStream servletOutputStream = response.raw().getOutputStream();
+          InputStream resultStream = content.getInputStream()) {
+        if (shouldGzip) {
+          response.header(HttpHeaders.CONTENT_ENCODING, "gzip");
+          LOGGER.trace("Request header accepts gzip");
+          try (OutputStream gzipServletOutputStream = new GZIPOutputStream(servletOutputStream)) {
+            IOUtils.copy(resultStream, gzipServletOutputStream);
+          }
+        } else {
+          IOUtils.copy(resultStream, servletOutputStream);
+        }
+      }
+    } catch (java.io.IOException e) {
+      LOGGER.debug("Error attaching output file", e);
+      return;
+    }
+
+    response.status(200);
+    LOGGER.trace("Response sent in {} file format.", fileExt);
```

Comment:
✏️ You have the queryResponseTransformer in this context so you could add it's `id` to this log statement.